src/main/java/org/radarcns/stream/statistics/SourceStatisticsStream.java

```diff
@@ -0,0 +1,247 @@
+
+import io.confluent.kafka.streams.serdes.avro.GenericAvroDeserializer;
+import io.confluent.kafka.streams.serdes.avro.SpecificAvroSerde;
+import io.confluent.kafka.streams.serdes.avro.SpecificAvroSerializer;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.processor.Cancellable;
+import org.apache.kafka.streams.processor.Processor;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.codehaus.jackson.annotate.JsonCreator;
+import org.codehaus.jackson.annotate.JsonProperty;
+import org.radarcns.config.SourceStatisticsMonitorConfig;
+import org.radarcns.kafka.ObservationKey;
+import org.radarcns.stream.AbstractStreamWorker;
+import org.radarcns.stream.SourceStatistics;
+import org.radarcns.stream.StreamDefinition;
+import org.radarcns.util.serde.RadarSerde;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SourceStatisticsStream extends AbstractStreamWorker {
+    private static final Logger logger = LoggerFactory.getLogger(SourceStatisticsStream.class);
+    private String name;
+    private Duration interval;
+
+    @Override
+    protected List<KafkaStreams> createStreams() {
+        return Collections.singletonList(new KafkaStreams(getTopology(), getStreamsConfig()));
+    }
+
+    @Override
+    protected void doCleanup() {
+        // do nothing
+    }
+
+    @Override
+    protected void initialize() {
+        SourceStatisticsMonitorConfig config = (SourceStatisticsMonitorConfig) this.config;
+
+        this.name = config.getName();
+
+        List<String> inputTopics = config.getTopics();
+        if (inputTopics == null || inputTopics.isEmpty()) {
+            throw new IllegalArgumentException("Input topics for stream " + name + " is empty");
+        }
+        if (config.getOutputTopic() == null) {
+            throw new IllegalArgumentException("Output topic for stream " + name + " is missing");
+        }
+        inputTopics.forEach(t -> defineStream(t, config.getOutputTopic()));
+
+        this.interval = Duration.ofMillis(config.getFlushTimeout());
+    }
+
+    protected Duration getInterval() {
+        return interval;
+    }
+
+    @SuppressWarnings("PMD.OptimizableToArrayCall")
+    private Topology getTopology() {
+        Topology builder = new Topology();
+        GenericAvroDeserializer genericReader = new GenericAvroDeserializer();
```

Comment:
What happens here if the schema id of a payload does not exist in the schema registry ?

Reply:
Updated default exception handler in KafkaProperty with `props.put(StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG,
                LogAndContinueExceptionHandler.class.getName());`

Reply:
Great. This also helps in other cases as mentioned in #71